tests/system/src/test/java/test/com/sun/prism/impl/NGNodeDirtyFlagTest.java

```diff
@@ -0,0 +1,190 @@
+
+    public static class MyApp extends Application {
+
+        private StackPane root;
+
+        public MyApp() {
+            super();
+        }
+
+        @Override
+        public void init() {
+            myApp = this;
+        }
+
+        @Override
+        public void start(Stage primaryStage) throws Exception {
+            root = new StackPane();
+            primaryStage.setScene(new Scene(root, 500, 400));
+
+            primaryStage.setOnShown(e -> Platform.runLater(launchLatch::countDown));
+            primaryStage.show();
+        }
+    }
+
+    @BeforeClass
+    public static void setupOnce() {
+        Util.launch(launchLatch, MyApp.class);
+    }
+
+    @AfterClass
+    public static void teardownOnce() {
+        Util.shutdown();
+    }
+
+    @Test
+    public void testNGNodesNotDirty() throws InterruptedException {
+        ObjectProperty<Color> lineColor = new SimpleObjectProperty<>(Color.DARKGREEN);
+        ObjectProperty<Color> circleColor = new SimpleObjectProperty<>(Color.DARKGREEN);
+
+        StackPane root = myApp.root;
+
+        Util.runAndWait(() -> {
+            var contents = new HBox();
+            contents.setSpacing(10);
+            contents.setPadding(new Insets(10));
+            contents.getChildren().add(contentElement("L", lineColor, circleColor));
+            contents.getChildren().add(contentElement("R", lineColor, circleColor));
+            root.getChildren().add(contents);
+
+            Pane sideArea = createSideArea();
+            StackPane.setAlignment(sideArea, Pos.CENTER_RIGHT);
+            root.getChildren().add(sideArea);
+        });
+
+        Util.waitForIdle(root.getScene());
+
+        for (int i = 0; i < 5; i++) {
+            Util.runAndWait(() -> lineColor.set(Color.LIGHTGREEN));
+            Util.waitForIdle(root.getScene());
+            Util.runAndWait(() -> circleColor.set(Color.LIGHTGREEN));
+            Util.waitForIdle(root.getScene());
+
+            checkLineColor(root, lineColor.get());
+
+            Util.runAndWait(() -> lineColor.set(Color.DARKGREEN));
+            Util.waitForIdle(root.getScene());
+            Util.runAndWait(() -> circleColor.set(Color.DARKGREEN));
+            Util.waitForIdle(root.getScene());
+
+            checkLineColor(root, lineColor.get());
+        }
+
+    }
+
+    private void checkLineColor(StackPane root, Color expected) {
+        Util.runAndWait(() -> {
+            checkColor(root.lookup("#Line-L"), expected);
+            checkColor(root.lookup("#Line-R"), expected);
+        });
+    }
+
+    private void checkColor(Node node, Color expected) {
+        Robot robot = new Robot();
+        Bounds screenBounds = node.localToScreen(node.getBoundsInLocal());
+        WritableImage image = robot.getScreenCapture(null, screenBounds.getMinX(), screenBounds.getMinY(), 100, 100);
+        Assert.assertEquals("A node was not rendered properly. Wrong color found", name(expected), name(image.getPixelReader().getColor(1, 1)));
```

Comment:
This method of color comparison seems convoluted and error prone. I recommend using one of the existing color comparison methods with toloerance. See VisualTestBase, for example.

Reply:
The test now uses ``VisualTestBase.assertColorEquals(Color expected, Color actual, double delta)`` to compare colors.